src/dbnode/storage/series/buffer.go

```diff
@@ -1256,7 +1269,13 @@ func (b *BufferBucket) write(
 		return err == nil, err
 	}
 
-	// Need a new encoder, we didn't find an encoder to write to
+	// Need a new encoder, we didn't find an encoder to write to.
+	maxEncoders := b.opts.RuntimeOptionsManager().Get().EncodersPerBlockLimit()
+	if maxEncoders != 0 && len(b.encoders) >= int(maxEncoders) {
+		b.opts.Stats().IncEncoderLimitWriteRejected()
+		return false, errTooManyEncoders
```

Comment:
one addition: a metric indicating a write was rejected due to this reason (can be done at this level) or you can make it ns/db level based on a check of an exported error, your call.

Reply:
Hm, that's what `IncEncoderLimitWriteRejected()` right above this line does. Or am I missing something?